sanic/blueprints.py

```diff
@@ -449,16 +449,11 @@ def event(self, event: str, timeout: Optional[Union[int, float]] = None):
 
     @staticmethod
     def _extract_value(*values):
-        value = values[-1]
-        for v in values:
-            if v is not None:
-                value = v
-                break
-        return value
+        return next((v for v in values if v is not None), values[-1])
```

Comment:
Is there a reason for using a generator instead of a list comprehension here?

Reply:
I think generators are better to be used when it's possible as it uses less memory since it doesn't build the whole list at once

Reply:
I am not sure that I agree with this logic here. Arguments are passed as an existing object (a tuple). They already are existing in memory in an iterable. Where are we rebuilding the list that we are incurring a memory overhead?

Reply:
yea, in this case here there's no memory overhead the main reason here for using generators is to use next instead of forloop